docs/cstor.md

```diff
@@ -0,0 +1,384 @@
+
+Note: It is common to reboot the nodes during Kubernetes upgrades and it may force the above situation if the nodes have local disks which are ephemeral in nature. It is NOT recommended to use OpenEBS cStor in such scenarios. OpenEBS 0.8.1 release is expected to have this support.
+
+**Delayed snapshots**
+
+In cStor, snapshots are taken only when the volume replicas are in quorum. For example, as soon as the volume is provisioned on cStor, the volume will be in ready state but the quorum attainment may take  few minutes. Snapshot commands during this period will be delayed or queued till the volumes replicas attain quorum. The snapshot commands received by the target are also delayed when the cStor volume is marked read-only because of no-quorum
+
+
+
+## Troubleshooting areas
+
+Following are most commonly observed areas of troubleshooting
+
+1. **iSCSI tools are not installed or iSCSI.d service is not running**
+
+   **Symptom:**
+
+   If iSCSI tools are not installed on the nodes, after lauching the application with PVC pointing to OpenEBS provisioner,  you may observe the application is in "ContainerCreating" state. 
+
+   `kubectl describe pod <app pod> ` may give an error that looks like the following
+
+   ```
+   MountVolume.WaitForAttach failed for volume "pvc-33408bf6-2307-11e9-98c4-000d3a028f9a" : executable file not found in $PATH
+   ```
+
+   **Resolution**: 
+
+   Install iSCSI tools and make sure iSCSI service is running. See [iSCSI installation](/docs/next/prerequisites.html#steps-for-configuring-and-verifying-open-iscsi)
+
+2. **Multi-attach error is seen in the logs**
+
+   **Symptom:**
+
+   In a node failure case, it is sometimes observed that the application pods that were running on the node that went down are moved, the cStor target pods are also moved to other nodes, but the application throws up an error similar to below:
+
+   ```
+   Warning  FailedAttachVolume  4m15s  attachdetach-controller            
+   Multi-Attach error for volume "pvc-d6d52184-1f24-11e9-bf1d-0a0c03d0e1ae" 
+   Volume is already exclusively attached to one node and can't be attached to another
+   ```
+
+   **Resolution ?**
+
+   This is because of the Kubernetes limitation [explained above](/docs/next/cstor.html#known-limitations). OpenEBS CSI driver will resolve this issue. See [roadmap](/docs/next/cstor.html#cstor-roadmap). 
+
+3. **Connect failed but reconnect succeeds**
+
+   [unable to connect and but then they go away after sometimes (get an example: unable to get disk path) ]
```

Comment:
Application is seen to be in the Container Creating state, with kubectl describe showing `Connection refused` error to the cStor PV. This error eventually gets rectified on the further retries, volume gets mounted and application is started. This error is seen in cases, where cStor target takes some time to initialize - do to either slow network speed of downloading and starting the cstor pod or because the cstor target is waiting for the replicas to connect and establish quorum.